core/ts.h

```diff
@@ -170,37 +194,175 @@ class TransitionSystem
    * states */
   bool no_next(const smt::Term & term) const;
 
+  // term building functionality -- forwards to the underlying SmtSolver
+  // assumes all Terms/Sorts belong to solver_
```

Comment:
We don't check this in smt-switch yet, but that's something I've been wondering if we should check. It might be nice to discuss this further. There's not an easy way to do this at the Pono level either. Terms/Sorts currently don't have a reference to the solver. So we could either add a reference to the solver in smt-switch or have a special function that needs to be implemented which allows you to check.